### DISCUSSION TITLE
Validation of Relations

### DISCUSSION THREAD

**Discussion by NilsBaumgartner1994** [directus/directus#15750] [Archive]
Consider this scenario:

Tables:
- Profile
   - user: Directus_user
- Device
   - profile: Profile
   - user: Directus_user
- Directus_user

What i want is that, whenever a user has a profile and wants to add a device to its profile, it should be checked.

My Idea was to check: Validation: device.profile.user === $CURRENT_USER

When trying to set this validation in the profile field in the device, it does not show me the related user.
<img width="855" alt="Bildschirmfoto 2022-09-26 um 10 19 49" src="https://user-images.githubusercontent.com/33295329/192227883-7c567c81-9797-4ae3-aa89-238e9601bd6a.png">

Same goes to the options at the permissions.
<img width="855" alt="Bildschirmfoto 2022-09-26 um 10 22 23" src="https://user-images.githubusercontent.com/33295329/192228355-5b9b570a-9bc3-4b5f-8fb8-fcf3a1d6e80a.png">

My last solution would be to write an own hook/extension for the table, but i would like to have it in the fontend.
Any suggestions?

**Comment by azrikahar** [comment]
> When trying to set this validation in the profile field in the device, it does not show me the related user.

This comment here should shed some light on why it was intentionally hidden: https://github.com/directus/directus/issues/15734#issuecomment-1257042718

> My last solution would be to write an own hook/extension for the table

A filter hook extension or a Flow with Filter event hook might be the potential solutions at the moment 👍
